File: task3/Java/Subtask3.java
import java.util.Scanner;

public class Subtask3 {
    public static void main(String[] args) {
        // Create a Scanner object to read input from the user
        Scanner scanner = new Scanner(System.in);

        // Prompt the user to enter the number of rows
        System.out.print("Enter the number of rows for the diamond pattern: ");
        int n = scanner.nextInt();
        scanner.close();

        // Generate and print the diamond pattern
        printDiamond(n);
    }

    public static void printDiamond(int n) {
        // Print the top half of the diamond
        for (int i = 0; i < n; i++) {
            // Print leading spaces
            for (int j = 0; j < n - i - 1; j++) {
                System.out.print(" ");
            }
            // Print asterisks
            for (int j = 0; j < 2 * i + 1; j++) {
                System.out.print("*");
            }
            // Move to the next line
            System.out.println();
        }

        // Print the bottom half of the diamond
        for (int i = n - 2; i >= 0; i--) {
            // Print leading spaces
            for (int j = 0; j < n - i - 1; j++) {
                System.out.print(" ");
            }
            // Print asterisks
            for (int j = 0; j < 2 * i + 1; j++) {
                System.out.print("*");
            }
            // Move to the next line
            System.out.println();
        }
    }
}

File: task3/C/Subtask-3.c
#include <stdio.h>

void printDiamond(int n) {
    int i, j;

    // Top half of the diamond
    for (i = 0; i < n; i++) {
        // Print spaces
        for (j = 0; j < n - i - 1; j++)
            printf(" ");
        // Print asterisks
        for (j = 0; j < 2 * i + 1; j++)
            printf("*");
        printf("\n");
    }

    // Bottom half of the diamond
    for (i = n - 2; i >= 0; i--) {
        // Print spaces
        for (j = 0; j < n - i - 1; j++)
            printf(" ");
        // Print asterisks
        for (j = 0; j < 2 * i + 1; j++)
            printf("*");
        printf("\n");
    }
}

int main() {
    int n;

    // Prompt user for input
    printf("Enter the number of rows for the diamond: ");
    scanf("%d", &n);

    // Print the diamond pattern
    printDiamond(n);

    return 0;
}
